### PR TITLE
Benchmark functions were locked into one of the tests

### PR DESCRIPTION
**Pull request by betatim**
Refactored benchmark functions into their own module and added
some basic (in)sanity tests for each.

### CODE REVIEW

File: skopt/tests/test_gp_opt.py [hunk 1]
```diff
@@ -3,25 +3,7 @@
...
-               P=10**-4 * np.asarray([[1312, 1696, 5569, 124, 8283, 5886],
-                                      [2329, 4135, 8307, 3736, 1004, 9991],
-                                      [2348, 1451, 3522, 2883, 3047, 6650],
-                                      [4047, 8828, 8732, 5743, 1091, 381]]),
-               A=np.asarray([[10, 3, 17, 3.50, 1.7, 8],
-                             [0.05, 10, 17, 0.1, 8, 14],
-                             [3, 3.5, 1.7, 10, 17, 8],
-                             [17, 8, 0.05, 10, 0.1, 14]])):
-    return -np.sum(alpha * np.exp(-np.sum(A * (x - P)**2, axis=1)))
+from skopt.benchmarks import branin, hartmann_6
```
Comment (glouppe) [inline skopt/tests/test_gp_opt.py]: one import per line

File: skopt/tests/test_benchmarks.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import numpy as np
+from sklearn.utils.testing import assert_array_almost_equal
+from sklearn.utils.testing import assert_almost_equal
+
+from skopt.benchmarks import branin, hartmann_6
```
Comment (glouppe) [inline skopt/tests/test_benchmarks.py]: same here

**Comment by glouppe** [comment]
Thanks, much better this way! +1 after the small changes on imports

**Comment by betatim** [comment]
This adds some of the functions mentioned in #16 unsure if we should close it or leave it open to attract more benchmark problems.

**Comment by betatim** [comment]
Reverted this in #30, will try again in a new PR.

**Pull request merged**